HomeTask5/MyNUnit/TestRunner.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Reflection;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+namespace MyNUnit
+{
+    /// <summary>
+    /// Class for MyNUnit Framework 
+    /// </summary>
+    public static class TestRunner
+    {
+        /// <summary>
+        /// Concurent collection to strore data about the tests
+        /// </summary>
+        public static BlockingCollection<TestStrcuct> MyTests { get; private set; }
+
+        /// <summary>
+        /// Printing the results of testing
+        /// </summary>
+        public static void PrintTestResults()
+        {
+            Console.WriteLine("Results");
+            foreach (var testResult in MyTests)
+            {
+                Console.WriteLine($"Method : {testResult.MethodInformation}; {(testResult.IsPassed ? "Passed in " + testResult.TimeConsumed:"")}" +
+                    $"{(testResult.IsFailed ? $"Failed + Expected: {testResult.Expected}, Got: {testResult.Got}":"")}" +
+                    $"{((testResult.IsIgnored)? "Ignored, message = " + testResult.Ignore_message:"")}");
+            }
+        }
+
+        /// <summary>
+        /// Loadign the dll and starting the tests, Cheks for not repeated dlls
+        /// </summary>
+        /// <param name="path">Path to folder, loading all the dlls in all the directorieds beneath it as well</param>
+        public static void Start(string path)
+        {
+            MyTests = new BlockingCollection<TestStrcuct>();
+            var dllFiles = Directory.GetFiles(path,"*.dll", SearchOption.AllDirectories);
+            var dllFilesNotRepeated = new HashSet<string>();
+            var downloadeddlls = new HashSet<string>();
+            foreach (var dll in dllFiles)
+            {
+                if (!downloadeddlls.Contains(dll.Split("\\")[^1]))
+                {
+                    downloadeddlls.Add(dll.Split("\\")[^1]);
+                    dllFilesNotRepeated.Add(dll);
+                }
+
+            }
+            foreach(var dll in dllFilesNotRepeated)
+            {
+                var assembly = Assembly.LoadFrom(dll);
+                var types = assembly.GetTypes();
+                Parallel.ForEach(types, TestStarter);
+            }
+        }
+
+        /// <summary>
+        /// Starting the all the tests with BeforeClass - Before - MyTest - After - AfterClass atrributes
+        /// </summary>
+        /// <param name="type">loaded assembly</param>
+        public static void TestStarter(Type type)
+        {
+            MethodsInvoker<BeforeClassAttribute>(type);
+            MethodsInvoker<MyTestAttribute>(type);
+            MethodsInvoker<AfterClassAttribute>(type);
+        }
+
+        /// <summary>
+        /// Invokes the methods with attribues, calling methods corresponding to the attribute type
```

Comment:
```suggestion
        /// Invokes the methods with attributes, calling methods corresponding to the attribute type
```
Вообще, столько опечаток --- это прямо очень плохо. Опечататься в комментарии не страшно, а вот не там поставить запятую в коде отделения ступени ракеты --- и ракета за миллионы долларов падает на ближайший город.